internal/api/impl/proxy/proxy.go

```diff
@@ -286,9 +291,60 @@ func (h *httpProxy) setupAuthentication(req *http.Request) error {
 		}
 		req.Header.Set("Authorization", fmt.Sprintf("%s %s", auth.Type, credential))
 	}
+	oauth := h.secret.OAuth
+	if oauth != nil {
+		token, err := h.getToken(req.Context(), oauth)
+		if err != nil {
+			return err
+		}
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.AccessToken))
+	}
+
 	return nil
 }
 
+// getToken exchanges the client credentials for an access token,
+// from the OAuth 2.0 provider.
+func (h *httpProxy) getToken(ctx context.Context, oauth *secretModel.OAuth) (*oauth2.Token, error) {
+	transport, err := h.prepareTransport()
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient := http.Client{
+		Transport: transport,
+	}
+
+	// add our http client with tls config
+	context.WithValue(ctx, oauth2.HTTPClient, httpClient)
```

Comment:
```suggestion
newContext := context.WithValue(ctx, oauth2.HTTPClient, httpClient)
```

Then you should use this new context when calling `conf.Token`